### ISSUE TITLE
Not succeed to install

### ISSUE THREAD

**Issue by juliensl** [exesse/mongodump-s3#148]
Hi everybody ! 

I am trying to install mongodump-s3 but I not suceed :'( 
I am on an Ubuntu 20.04, what I did : 

`pip3 install mongodump-s3`
`mongodump-s3 --help` and I have : 

```
$ mongodump-s3 --help
Traceback (most recent call last):
  File "/home/julien/.local/bin/mongodump-s3", line 5, in <module>
    from mongodump_s3.__main__ import main
  File "/home/julien/.local/lib/python3.8/site-packages/mongodump_s3/__init__.py", line 3, in <module>
    from .s3 import S3
  File "/home/julien/.local/lib/python3.8/site-packages/mongodump_s3/s3.py", line 8, in <module>
    from azure.storage.blob import BlobServiceClient as AzureClient
ImportError: cannot import name 'BlobServiceClient' from 'azure.storage.blob' (/usr/lib/python3/dist-packages/azure/storage/blob/__init__.py)
```

I checked my libraries : 

```
$ pip3 list
Package                                          Version             
------------------------------------------------ --------------------
adal                                             1.2.2               
aiohttp                                          3.6.2               
applicationinsights                              0.11.9              
apturl                                           0.5.2               
argcomplete                                      1.8.1               
async-timeout                                    3.0.1               
attrs                                            19.3.0              
azure                                            4.0.0               
azure-ai-textanalytics                           1.0.0b1             
azure-appconfiguration                           1.0.0               
azure-applicationinsights                        0.1.0               
azure-batch                                      8.0.0               
azure-cli                                        2.0.81              
azure-cli-core                                   2.0.81              
azure-cli-telemetry                              1.0.4               
azure-cognitiveservices-anomalydetector          0.2.0               
azure-cognitiveservices-formrecognizer           0.1.0               
azure-cognitiveservices-knowledge-qnamaker       0.1.0               
azure-cognitiveservices-language-luis            0.5.0               
azure-cognitiveservices-language-spellcheck      2.0.0               
azure-cognitiveservices-language-textanalytics   0.2.0               
azure-cognitiveservices-personalizer             0.1.0               
azure-cognitiveservices-search-autosuggest       0.2.0               
azure-cognitiveservices-search-customimagesearch 0.2.0               
azure-cognitiveservices-search-customsearch      0.3.0               
azure-cognitiveservices-search-entitysearch      2.0.0               
azure-cognitiveservices-search-imagesearch       2.0.0               
azure-cognitiveservices-search-newssearch        2.0.0               
azure-cognitiveservices-search-videosearch       2.0.0               
azure-cognitiveservices-search-visualsearch      0.2.0               
azure-cognitiveservices-search-websearch         2.0.0               
azure-cognitiveservices-vision-computervision    0.5.0               
azure-cognitiveservices-vision-contentmoderator  1.0.0               
azure-cognitiveservices-vision-customvision      1.0.0               
azure-cognitiveservices-vision-face              0.4.0               
azure-common                                     1.1.24              
azure-core                                       1.16.0              
azure-cosmos                                     3.1.1               
azure-cosmosdb-table                             1.0.6               
azure-datalake-store                             0.0.48              
azure-devops                                     0.17.0              
azure-eventgrid                                  1.3.0               
azure-eventhub                                   5.0.0               
azure-eventhub-checkpointstoreblob               1.0.0               
azure-eventhub-checkpointstoreblob-aio           1.0.0               
azure-functions-devops-build                     0.0.22              
azure-graphrbac                                  0.61.1              
azure-identity                                   1.2.0               
azure-keyvault                                   4.0.0               
azure-keyvault-certificates                      4.0.0               
azure-keyvault-keys                              4.1.0b1             
azure-keyvault-secrets                           4.1.0b1             
azure-loganalytics                               0.1.0               
azure-mgmt                                       5.0.0a1             
azure-mgmt-advisor                               3.0.0               
azure-mgmt-alertsmanagement                      0.2.0rc2            
azure-mgmt-apimanagement                         0.1.0               
azure-mgmt-appconfiguration                      0.3.0               
azure-mgmt-applicationinsights                   0.2.0               
azure-mgmt-appplatform                           0.1.0               
azure-mgmt-attestation                           0.1.0               
azure-mgmt-authorization                         0.60.0              
azure-mgmt-automation                            0.1.1               
azure-mgmt-azurestack                            0.1.0               
azure-mgmt-batch                                 7.0.0               
azure-mgmt-billing                               0.2.0               
azure-mgmt-botservice                            0.2.0               
azure-mgmt-cdn                                   4.1.0rc1            
azure-mgmt-cognitiveservices                     5.0.0               
azure-mgmt-commerce                              1.0.1               
azure-mgmt-compute                               10.0.0              
azure-mgmt-consumption                           3.0.0               
azure-mgmt-containerinstance                     1.5.0               
azure-mgmt-containerregistry                     3.0.0rc8            
azure-mgmt-containerservice                      8.2.0               
azure-mgmt-cosmosdb                              0.11.0              
azure-mgmt-costmanagement                        0.1.0               
azure-mgmt-databoxedge                           0.1.0               
azure-mgmt-databricks                            0.1.0               
azure-mgmt-datafactory                           0.8.0               
azure-mgmt-datalake-analytics                    0.6.0               
azure-mgmt-datalake-store                        0.5.0               
azure-mgmt-datamigration                         4.0.0               
azure-mgmt-datashare                             0.1.0rc1            
azure-mgmt-deploymentmanager                     0.2.0               
azure-mgmt-devspaces                             0.2.0               
azure-mgmt-devtestlabs                           4.0.0               
azure-mgmt-dns                                   3.0.0               
azure-mgmt-documentdb                            0.1.3               
azure-mgmt-edgegateway                           0.1.0               
azure-mgmt-eventgrid                             3.0.0rc4            
azure-mgmt-eventhub                              3.0.0               
azure-mgmt-frontdoor                             0.3.0               
azure-mgmt-hanaonazure                           0.12.0              
azure-mgmt-hdinsight                             1.4.0               
azure-mgmt-healthcareapis                        0.1.0               
azure-mgmt-hybridcompute                         0.1.1               
azure-mgmt-imagebuilder                          0.3.0               
azure-mgmt-iotcentral                            2.0.0               
azure-mgmt-iothub                                0.10.0              
azure-mgmt-iothubprovisioningservices            0.2.0               
azure-mgmt-keyvault                              2.0.0               
azure-mgmt-kusto                                 0.5.0               
azure-mgmt-labservices                           0.1.1               
azure-mgmt-loganalytics                          0.2.0               
azure-mgmt-logic                                 4.0.0rc2            
azure-mgmt-machinelearningcompute                0.4.1               
azure-mgmt-machinelearningservices               0.1.0               
azure-mgmt-maintenance                           0.1.0               
azure-mgmt-managedservices                       1.0.0               
azure-mgmt-managementgroups                      0.2.0               
azure-mgmt-managementpartner                     0.1.1               
azure-mgmt-maps                                  0.1.0               
azure-mgmt-marketplaceordering                   0.2.1               
azure-mgmt-media                                 2.1.0               
azure-mgmt-mixedreality                          0.1.0               
azure-mgmt-monitor                               0.7.0               
azure-mgmt-msi                                   1.0.0               
azure-mgmt-netapp                                0.8.0               
azure-mgmt-network                               9.0.0               
azure-mgmt-notificationhubs                      2.1.0               
azure-mgmt-operationsmanagement                  0.1.0               
azure-mgmt-peering                               0.1.0rc2            
azure-mgmt-policyinsights                        0.4.0               
azure-mgmt-powerbidedicated                      0.1.0               
azure-mgmt-powerbiembedded                       2.0.0               
azure-mgmt-privatedns                            0.1.0               
azure-mgmt-rdbms                                 1.9.0               
azure-mgmt-recoveryservices                      0.4.0               
azure-mgmt-recoveryservicesbackup                0.6.0               
azure-mgmt-redis                                 7.0.0rc1            
azure-mgmt-relay                                 0.2.0               
azure-mgmt-reservations                          0.7.0               
azure-mgmt-resource                              8.0.0               
azure-mgmt-resourcegraph                         2.0.0               
azure-mgmt-scheduler                             2.0.0               
azure-mgmt-search                                2.1.0               
azure-mgmt-security                              0.3.0               
azure-mgmt-serialconsole                         0.1.0               
azure-mgmt-servermanager                         2.0.0               
azure-mgmt-servicebus                            0.6.0               
azure-mgmt-servicefabric                         0.4.0               
azure-mgmt-signalr                               0.3.0               
azure-mgmt-sql                                   0.16.0              
azure-mgmt-sqlvirtualmachine                     0.5.0               
azure-mgmt-storage                               7.1.0               
azure-mgmt-storagecache                          0.2.0               
azure-mgmt-storagesync                           0.2.0               
azure-mgmt-subscription                          0.5.0               
azure-mgmt-timeseriesinsights                    0.1.0               
azure-mgmt-trafficmanager                        0.51.0              
azure-mgmt-vmwarecloudsimple                     0.2.0               
azure-mgmt-web                                   0.44.0              
azure-multiapi-storage                           0.2.4               
azure-servicebus                                 0.50.2              
azure-servicefabric                              7.0.0.0             
azure-servicemanagement-legacy                   0.20.7              
azure-storage-blob                               12.6.0              
azure-storage-common                             1.4.0               
azure-storage-file                               1.4.0               
azure-storage-file-datalake                      12.0.0b7            
azure-storage-file-share                         12.0.1              
azure-storage-queue                              1.4.0               
azure-template                                   0.0.1               
bcrypt                                           3.1.7               
blinker                                          1.4                 
boto3                                            1.16.50             
botocore                                         1.19.63             
Brlapi                                           0.7.0               
cachetools                                       4.2.2               
certifi                                          2019.11.28          
cffi                                             1.14.0              
chardet                                          3.0.4               
Click                                            7.0                 
colorama                                         0.4.3               
command-not-found                                0.3                 
cryptography                                     2.8                 
cupshelpers                                      1.0                 
dbus-python                                      1.2.16              
defer                                            1.0.6               
distro                                           1.4.0               
distro-info                                      0.23ubuntu1         
duplicity                                        0.8.12.0            
entrypoints                                      0.3                 
fabric                                           2.5.0               
fasteners                                        0.14.1              
future                                           0.18.2              
google-api-core                                  1.31.0              
google-auth                                      1.32.1              
google-cloud-core                                1.7.1               
google-cloud-storage                             1.35.0              
google-crc32c                                    1.1.2               
google-resumable-media                           1.3.1               
googleapis-common-protos                         1.53.0              
httplib2                                         0.14.0              
humanfriendly                                    4.18                
hurry.filesize                                   0.9                 
idna                                             2.8                 
invoke                                           1.3.0               
isodate                                          0.6.0               
javaproperties                                   0.6.0               
Jinja2                                           2.10.1              
jmespath                                         0.9.4               
jsmin                                            2.2.2               
jsondiff                                         1.1.1               
keyring                                          18.0.1              
knack                                            0.6.3               
language-selector                                0.1                 
launchpadlib                                     1.10.13             
lazr.restfulclient                               0.14.2              
lazr.uri                                         1.0.3               
lockfile                                         0.12.2              
louis                                            3.12.0              
macaroonbakery                                   1.3.1               
Mako                                             1.1.0               
MarkupSafe                                       1.1.0               
mock                                             3.0.5               
mongodump-s3                                     1.1.0               
monotonic                                        1.5                 
msal                                             1.1.0               
msal-extensions                                  0.1.3               
msrest                                           0.6.10              
msrestazure                                      0.6.2               
multidict                                        4.7.3               
netifaces                                        0.10.4              
oauthlib                                         3.1.0               
olefile                                          0.46                
packaging                                        21.0                
paramiko                                         2.6.0               
pbr                                              5.4.5               
pexpect                                          4.6.0               
Pillow                                           7.0.0               
pip                                              20.0.2              
pkginfo                                          1.4.2               
ply                                              3.11                
portalocker                                      1.5.1               
protobuf                                         3.17.3              
psutil                                           5.5.1               
pyasn1                                           0.4.8               
pyasn1-modules                                   0.2.8               
pycairo                                          1.16.2              
pycparser                                        2.19                
pycups                                           1.9.73              
Pygments                                         2.3.1               
PyGObject                                        3.36.0              
PyJWT                                            1.7.1               
pymacaroons                                      0.13.0              
PyNaCl                                           1.3.0               
pyOpenSSL                                        19.0.0              
pyparsing                                        2.4.7               
pyRFC3339                                        1.1                 
python-apt                                       2.0.0+ubuntu0.20.4.5
python-dateutil                                  2.7.3               
python-debian                                    0.1.36ubuntu1       
python-dotenv                                    0.15.0              
pytz                                             2019.3              
pyxdg                                            0.26                
PyYAML                                           5.3.1               
reportlab                                        3.5.34              
requests                                         2.25.1              
requests-oauthlib                                1.0.0               
requests-unixsocket                              0.2.0               
rsa                                              4.7.2               
s3transfer                                       0.3.7               
scp                                              0.13.0              
SecretStorage                                    2.3.1               
setuptools                                       57.1.0              
simplejson                                       3.16.0              
six                                              1.14.0              
sshtunnel                                        0.1.4               
systemd-python                                   234                 
tabulate                                         0.8.6               
uamqp                                            1.2.6               
ubuntu-advantage-tools                           27.0                
ubuntu-drivers-common                            0.0.0               
ufw                                              0.36                
unattended-upgrades                              0.1                 
urllib3                                          1.25.8              
usb-creator                                      0.3.7               
vsts-cd-manager                                  1.0.2               
wadllib                                          1.3.3               
websocket-client                                 0.53.0              
wheel                                            0.36.2              
xkit                                             0.0.0               
xmltodict                                        0.12.0              
yarl                                             1.4.2
```

I think I have the good version of Azure-storage-blob, I don't understand 🤔  

Does anybody have an idea ? 

Thank you for your help !

**Comment by exesse** [comment]
@juliensl Hi, I will check later today. Maybe in meantime you would like to try the docker image of the service to run your task?

**Comment by exesse** [comment]
@juliensl please run 
```
pip3 install azure-storage-blob --upgrade
```
The package requires "azure-storage-blob==12.7.1" and you have 12.6.0.

Hope this will solve the issue.

**Comment by juliensl** [comment]
Hi @exesse , thank you for your answer.

I tried to change the version, using the last version 12.8.1 or using the specific version 12.7.1 as you recommended me. Both does not work : 

`ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
mongodump-s3 1.1.0 requires azure-storage-blob==12.6.0, but you have azure-storage-blob 12.8.1 which is incompatible.`

`ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
mongodump-s3 1.1.0 requires azure-storage-blob==12.6.0, but you have azure-storage-blob 12.7.1 which is incompatible.`

When I try to run mongodump-s3 : 

```
Traceback (most recent call last):
  File "/home/julien/.local/bin/mongodump-s3", line 5, in <module>
    from mongodump_s3.__main__ import main
  File "/home/julien/.local/lib/python3.8/site-packages/mongodump_s3/__init__.py", line 3, in <module>
    from .s3 import S3
  File "/home/julien/.local/lib/python3.8/site-packages/mongodump_s3/s3.py", line 8, in <module>
    from azure.storage.blob import BlobServiceClient as AzureClient
ImportError: cannot import name 'BlobServiceClient' from 'azure.storage.blob' (/usr/lib/python3/dist-packages/azure/storage/blob/__init__.py)
```

Thank you for your help

**Comment by exesse** [comment]
Hi @juliensl,

I have updated the dependencies and released a new version 1.1.2 of the tool. Please run:

```
python3 -m pip install mongodump-s3 --upgrade
```

Please let me know if this solved the issue for you.
